bofire/data_models/features/tasks.py

```diff
@@ -0,0 +1,32 @@
+from typing import List, Literal
+
+import numpy as np
+from pydantic import model_validator, validator
+
+from bofire.data_models.features.api import DiscreteInput
+
+
+class TaskInput(DiscreteInput):
```

Comment:
Short question: why are you basing it on the `DiscreteInput` and not on the `CategoricalInput`? I would prefer the `CategoricalInput`, as it is more expressive and the Task is somehow more a categorical quantity than a discrete one, as the `DiscreteInput` always asumes an ordinal ordering of its values.

Reply:
I thought about it and here is my logic:

(a) The task themselves do not have any ordering

(b) But the GP expects an ordering of the tasks corresponding to the location of each specific task in the IndexKernel function, i.e. inputs to kernel are `k((x', i), (x, j)) ` so the task input values should always correspond to values `i, j in [0, ..., n_tasks - 1]`. 

This means that all we need to do to initialize the object InputTasks is to select the number of tasks. I am happy to change this though, and we can just encode the Categorical variables before feeding it to the GP.

Reply:
It is now using `CategoricalInput` as a base